skyportal/tests/rate_limiting/test_rate_limiting.py

```diff
@@ -0,0 +1,55 @@
+import time
+import socket
+import requests
+from baselayer.app.env import load_env
+from skyportal.tests import api
+
+
+_, cfg = load_env()
+
+s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+s.connect(("8.8.8.8", 53))
+localhost_external_ip = s.getsockname()[0]
+s.close()
+
+
+def test_api_rate_limiting(view_only_token):
+    # In case this test gets run after those defined below
+    time.sleep(1)
+    for n_successful_requests in range(100):
+        r = requests.get(
+            f'http://{localhost_external_ip}:{cfg["ports.app"]}/api/sysinfo',
+            headers={'Authorization': f'token {view_only_token}'},
+        )
+        if r.status_code != 200:
+            break
+    assert 14 <= n_successful_requests <= 16
+    r = requests.get(
+        f'http://{localhost_external_ip}:{cfg["ports.app"]}/api/sysinfo',
+        headers={'Authorization': f'token {view_only_token}'},
+    )
+    assert r.status_code == 429
+
+    time.sleep(1)
+
+    r = requests.get(
+        f'http://{localhost_external_ip}:{cfg["ports.app"]}/api/sysinfo',
+        headers={'Authorization': f'token {view_only_token}'},
+    )
+    assert r.status_code == 200
+
+
+def test_rate_limited_requests_ok(view_only_token):
+    for i in range(30):
+        r = requests.get(
+            f'http://{localhost_external_ip}:{cfg["ports.app"]}/api/sysinfo',
+            headers={'Authorization': f'token {view_only_token}'},
+        )
+        assert r.status_code == 200
+        time.sleep(0.2)
+
+
+def test_localhost_unlimited(view_only_token):
+    for i in range(30):
```

Comment:
Left a comment here, but somehow it disappeared.

Curious what this is testing if the number is the same as the above.  Shouldn't the number here be higher?

Reply:
This test is un-throttled (all requests as quickly as possible) and is coming from localhost (note we're not using the external IP address in this test). In the test above, we are using the external IP address but _throttling_ down to at most 5/s.

Reply:
See test names

Reply:
That makes perfect sense; can you add a comment along those lines, just to help readers along?  I couldn't quite grasp that this was happening from the function names only.

After that, go ahead and merge once @kmshin1397 approves.  Thanks, Ari!